default.nix

```diff
@@ -1,12 +1,31 @@
-{ system ? builtins.currentSystem }:
+{ system ? builtins.currentSystem
+
+, compiler ? "ghc948"
+
+, # Commit of the Nixpkgs repository that we want to use.
+  nixpkgsVersion ? import nix/nixpkgs-version.nix
+
+, # Nix files that describe the Nixpkgs repository. We evaluate the expression
+  # using `import` below.
+  nixpkgs ? builtins.fetchTarball {
+    url = "https://github.com/nixos/nixpkgs/archive/${nixpkgsVersion.rev}.tar.gz";
+    sha256 = nixpkgsVersion.tarballHash;
+  }
+
+, applyPatches ? (
+    # If nixpkgs is a flake
+    nixpkgs.legacyPackages.${system}
+      or
+      # If nixpkgs is a derivation
+      import nixpkgs
+      { inherit system; }
+  ).applyPatches
```

Comment:
Ah, I hadn't realized that would cause CI issues — I'd guess it's because the patch source pulls in the entire repo?

In any case, it's plausible there will be future cases where nixpkgs needs to be patched, even if temporarily. This at least avoids one extra import (if using flakes).

Reply:
> I guess the proper approach here would be to temporarily point nixpkgs at e.g. my fork where the patch is applied already. Not sure how that would interact with flakes, though.

With flakes it's straightforward. Actually, this could be done now — keep a postgrest fork of nixpkgs and refer to that in nixpkgsVersions etc. Then, with a flake, one would set nixpkgs to the upstream version or the fork. Stepping back further, a better approach would be for the Haskell-wrapper to be first class so one could override it.

> This is to avoid downloading stuff from cachix which doesn't need to be rebuild.

This is a bit of an annoyance. I think I've used [`nix-build-uncached`](https://github.com/Mic92/nix-build-uncached) in the past but it's been a while since I've needed to reach for this sort of thing.

> But the patched nixpkgs is not cached in cachix (actually not sure why) and thus causes full downloads of all packages depending on it.

I think this is one of those "quit thinking and look" cases… that said, it wouldn't surprise me if the derivation for the patched nixpkgs is depending on this repo source. If so, this PR might fix that, as I removed the `overlays` from the `import` of `nixpkgs` that defines `applyPatches`.